src/memory/query_memory_control.cpp

```diff
@@ -102,6 +102,17 @@ void QueriesMemoryControl::InitializeArenaCounter(unsigned arena_ind) {
   arena_tracking[arena_ind].store(0, std::memory_order_relaxed);
 }
 
+void QueriesMemoryControl::ResetTrackings() {
+  // First disable trackings on arenas
+  // so we don't start tracking on some query if
+  // there was something active left
+  for (auto &[arena_id, counter] : arena_tracking) {
+    counter.store(0, std::memory_order_acquire);
+  }
+  transaction_id_to_tracker.clear();
+  thread_id_to_transaction_id.clear();
```

Comment:
What if a `RemoveTrackingOnArena` happens making a counter negative.
I'm not confident the behavior of this will be correct if `FreeMemory` called at any arbitrary point during a query execution.

Reply:
So this is completely outdated logic, I removed that part completely on different diff, so I will resolve this